### PR TITLE
Added tests for LoggerMessage.Define with skipEnabledCheck

### PR DESCRIPTION
**Pull request by gfoidl**
Cf. https://github.com/dotnet/runtime/pull/50334#issuecomment-812177319

/cc: @maryamariyan

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Logging/tests/DI.Common/Common/src/TestLogger.cs [hunk 1]
```diff
@@ -41,11 +41,6 @@ public IDisposable BeginScope<TState>(TState state)
 
         public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
         {
-            if (!IsEnabled(logLevel))
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Logging/tests/DI.Common/Common/src/TestLogger.cs:44]: would this not break any existing test?

Comment (gfoidl) [reply src/libraries/Microsoft.Extensions.Logging/tests/DI.Common/Common/src/TestLogger.cs:44]: No existing tests broke.

The reason for that change, is to see if `skipEnabledCheck` works as https://github.com/dotnet/runtime/blob/311d37de4750d856f3e2292fdf8bacc1bcc62709/src/libraries/Microsoft.Extensions.Logging/tests/Common/LoggerMessageTest.cs#L202 shouldn't be logged if set to `false`, but should be logged when set to `true`. 
If `Log`-method would check, then it won't be logged, thus it's removed.

A different approach would be to call-count `IsEnabled` and assert that value.

Comment (maryamariyan) [reply src/libraries/Microsoft.Extensions.Logging/tests/DI.Common/Common/src/TestLogger.cs:44]: I see, I think call-counting `IsEnabled` and asserting sounds a bit clearer for future reference.

Comment (gfoidl) [reply src/libraries/Microsoft.Extensions.Logging/tests/DI.Common/Common/src/TestLogger.cs]: Changed to call-counting -> d1b4e93652016f93eceafcadbe1ac1f0b0ae3de5

**Review by maryamariyan — APPROVED** [review]
LGTM. Thanks @gfoidl

**Comment by safern** [comment]
Closed and re-opened to kick CI again.

**Pull request merged**